generate/create_clf_data.py

```diff
@@ -75,44 +78,45 @@ def create_clf_data(dataset_name, dataset_path, image_name, image, data):
 
 def convert_data(dataset_type, num, offset=0):
 
-    new_dataset = 'clf_' + dataset_type
+    new_dataset = ('clf_' + dataset_type, ) * num # Broadcast our data to an num-len tuple for multithreading
     images_path = os.path.join(config.DATA_DIR, dataset_type, 'images')
-    new_images_path = os.path.join(config.DATA_DIR, new_dataset, 'images')
+    new_images_path = (os.path.join(config.DATA_DIR, new_dataset[0], 'images'), ) * num
```

Comment:
If for some reason if `config.NUM_IMAGES` or `config.NUM_VAL_IMAGES` is 0, this line will give us an error because we are indexing an empty list.

Reply:
Fixed